pkg/frontend/validate.go

```diff
@@ -44,3 +44,22 @@ func (f *frontend) validateSubscriptionState(ctx context.Context, key string, al
 
 	return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidSubscriptionState, "", "Request is not allowed in subscription in state '%s'.", doc.Subscription.State)
 }
+
+// validateOpenShiftUniqueKey returns which unique key if causing a 412 error
+func (f *frontend) validateOpenShiftUniqueKey(ctx context.Context, doc *api.OpenShiftClusterDocument) error {
+	docs, err := f.db.OpenShiftClusters.ValidateUniqueKey(ctx, doc.PartitionKey, "clusterResourceGroupIdKey", doc.ClusterResourceGroupIDKey)
```

Comment:
Not sure if having the key name "hardcoded" like this is very clean. Was trying to have 1 function to handle both cases